### DISCUSSION TITLE
Road to OKD 4.10

### DISCUSSION THREAD

**Discussion by vrutkovs** [okd-project/okd#995] [Releases]
* [x] ~~Branch machine-config-operator in release-4.10-okd~~
       Fork is no longer required

* [x] Branch okd-machine-os in release-4.10

* [x] Create release-controller configuration
       https://github.com/openshift/release/pull/25808

* [x] Reconfigure CI to use there branches, create necessary imagestreams (can be completed later)

  * [x] cluster-samples-operator
  * [x] cluster-update-keys
  * [ ] [origin](https://github.com/openshift/release/pull/26163)
  * [ ] [origin-branding](https://github.com/openshift/release/pull/24060)
  * [x] [ovn-kubernetes](https://github.com/openshift/release/pull/26164)
  * [x] operator-marketplace

* [x] Update okd-machine-os release-4.10 branch to use stable fedora-coreos-config
        https://github.com/openshift/okd-machine-os/pull/296

* [x] Setup upgrade readiness jobs
	   https://github.com/openshift/release/pull/26165

* [x] Monitor [upgrade job](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-okd-4.10-upgrade-from-okd-4.9-e2e-upgrade-aws)
	* [x] [Successful upgrade](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-aws-modern/1499682109839642624) 

* [x] Fix found upgrade issues
	* [x] 	https://bugzilla.redhat.com/show_bug.cgi?id=2058729

* [x] Wait for first OCP 4.10 to be released

* [x] [Rebase okd-machine-os](https://github.com/openshift/okd-machine-os/pull/305)

* [x] Release a new stable

* [ ] [Bump FCOS version in openshift-installer](https://github.com/openshift/installer/pull/5680)

* [ ] Docs: add a note that BM IPI is now available

* [x] Update Assisted Instaler okd-configmap to use OKD 4.10

**Comment by alibo** [comment]
Is it also possible to upgrade directly from 4.8 to 4.10?

**Reply by vrutkovs** [reply]
No, your control plane needs to be updated from 4.8 to 4.9 to 4.10. You can upgrade workers from 4.8 to 4.10 directly though - see https://docs.openshift.com/container-platform/4.8///updating/preparing-eus-eus-upgrade.html

**Comment by bdlink** [comment]
Resolving the CEPH issue in a way that does not require recreating PVs and PVCs is important for upgrades.  Perhaps that should be added here?

**Reply by vrutkovs** [reply]
That would require us downgrading the kernel to 5.15, which is too significant imo. Lets discuss it in the bug - if the workaround is too complicated for most users we may consider it

**Reply by fortinj66** [reply]
unfortunately the 'fix' is destructive to the currently existing PVC/PVs...  I'll be backing up and volumes that have data I need to keep and then restoring after the fix.  

Only other option (besides downgrading the kernel) is to see if there is a kernel fix down the road.  That could be a long time coming though.

**Comment by vrutkovs** [comment]
[Road to 4.11](https://github.com/openshift/okd/discussions/1282)
